packages/edit-post/src/components/sidebar/post-template/create-modal.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * External dependencies
+ */
+import { kebabCase } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { store as editorStore } from '@wordpress/editor';
+import { useState } from '@wordpress/element';
+import { serialize, createBlock } from '@wordpress/blocks';
+import {
+	Modal,
+	Flex,
+	FlexItem,
+	TextControl,
+	Button,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { store as editPostStore } from '../../../store';
+
+const DEFAULT_TITLE = __( 'Custom Template' );
+
+export default function PostTemplateCreateModal( { onClose } ) {
+	const settings = useSelect(
+		( select ) => select( editorStore ).getEditorSettings(),
+		[]
+	);
```

Comment:
Suggestion: I would recommend only returning the setting (`defaultBlockTemplate`) needed for this component to avoid re-rendering if any other editor's setting changes.

Reply:
I had to do this anyway cause splitting `useSelect` up was causing a bug due to calling `getEditedPostTemplate` on non block themes! 😀